docker-compose-rule-core/src/main/java/com/palantir/docker/compose/connection/waiting/ClusterHealthCheck.java

```diff
@@ -25,9 +25,13 @@
 import java.util.List;
 import java.util.Set;
 import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 @FunctionalInterface
 public interface ClusterHealthCheck {
+    Logger log = LoggerFactory.getLogger(ClusterHealthCheck.class);
```

Comment:
Having this here is sad. Should I make `ClusterHealthCheck ` an abstract class and have this as a private variable?

Reply:
Nope - instead, note that this is only used in NativeHealthCheck and do

```
static ClusterHealthCheck nativeHealthChecks() {
    return new ClusterHealthCheck() {
        static final Logger log = LoggerFactory.getLogger(ClusterHealthCheck.class);
        @Override
        SuccessOrFailure isClusterHealthy(Cluster cluster) throws InterruptedException {
            Set<String> unhealthyContainers = new LinkedHashSet<>();
            .........
        }
    }
}
```